### PR TITLE
Replaced broken links

### LINKED ISSUE #679: Clean-up broken links
Running a link checker has shown up a bunch of broken links on the site:

- https://developer.mozilla.org/docs/Tools/Remote_Debugging/Debugging_Firefox_for_Android_with_WebIDE from /documentation/develop/differences-between-desktop-and-android-extensions/
- https://developer.mozilla.org/docs/Tools/Remote_Debugging/Debugging_Firefox_for_Android_with_WebIDE#Enable_remote_debugging_in_Firefox_for_Android from /documentation/develop/developing-extensions-for-firefox-for-android/
- https://developer.mozilla.org/docs/browser_specific_settings from /documentation/develop/test-permission-requests/ (2 cases)

17 Github username links across 55 instances:

- https://github.com/mdnwebdocs-bot
- https://github.com/hatem45
- https://github.com/david_ross
- https://github.com/potterwrit
- https://github.com/andymckay-github
- https://github.com/V@no
- https://github.com/hasangol
- https://github.com/Cikgu77
- https://github.com/01787500664
- https://github.com/tedmcox
- https://github.com/iwanrezpect
- https://github.com/gray_-_wolf
- https://github.com/Leif-AMO
- https://github.com/johnadungan
- https://github.com/PikadudeNo1
- https://github.com/niharikak101
- https://github.com/Zearin_Galaurum

### PR DESCRIPTION
**Pull request by Yashs911**
Fixes #679

### CODE REVIEW

**Review by muffinresearch — CHANGES_REQUESTED** [review]
@Yashs911 thanks for the patch. Looks good - just one small nit with the locale in the URL

File: src/content/documentation/develop/test-permission-requests.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ You can use an HTTP server (such as a simple [Python localhost server](https://d
 To perform the test you'll need to:
 
 - determine the address of the HTTP or HTTPS server where you can host files.
-- use the `manifest.json` [`browser_specific_settings`](https://developer.mozilla.org/docs/browser_specific_settings) key to:
+- use the `manifest.json` [`browser_specific_settings`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) key to:
```
[the remark this answers is not in the document]
Comment (Yashs911) [reply src/content/documentation/develop/test-permission-requests.md:81]: @muffinresearch okay

**Review by muffinresearch — APPROVED** [review]
Looks good, thanks!

**Comment by caitmuenster** [comment]
Thanks so much, @Yashs911, and welcome onboard! 🙌 Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition).

**Pull request merged**